examples/widgets/scale/lv_example_scale_7.c

```diff
@@ -0,0 +1,65 @@
+#include "../../lv_examples.h"
+#if LV_USE_SCALE && LV_BUILD_EXAMPLES
+
+#include "../../../src/lvgl_private.h" //To expose the fields of lv_draw_task_t
+
+static void draw_event_cb(lv_event_t * e)
+{
+    lv_obj_t * obj = lv_event_get_target(e);
+    lv_draw_task_t * draw_task = lv_event_get_draw_task(e);
+    lv_draw_dsc_base_t * base_dsc = lv_draw_task_get_draw_dsc(draw_task);
+    lv_draw_label_dsc_t * label_draw_dsc = lv_draw_task_get_label_dsc(draw_task);
+    if(base_dsc->part == LV_PART_INDICATOR) {
+        if(label_draw_dsc) {
+            const lv_color_t color_idx[7] = {
+                lv_palette_main(LV_PALETTE_RED),
+                lv_palette_main(LV_PALETTE_ORANGE),
+                lv_palette_main(LV_PALETTE_YELLOW),
+                lv_palette_main(LV_PALETTE_GREEN),
+                lv_palette_main(LV_PALETTE_CYAN),
+                lv_palette_main(LV_PALETTE_BLUE),
+                lv_palette_main(LV_PALETTE_PURPLE),
+            };
+            uint8_t major_tick = lv_scale_get_major_tick_every(obj);
+            label_draw_dsc->color = color_idx[base_dsc->id1 / major_tick];
+            lv_snprintf((char *)label_draw_dsc->text,
+                        sizeof(label_draw_dsc->text),
+                        "%.1f",
+                        base_dsc->id2 * 1.0f);
```

Comment:
I've just realized that it's incorrect this way.

```suggestion
            lv_snprintf((char *)label_draw_dsc->text,
                        lv_strlen(label_draw_dsc->text), /*Not the size of the pointer*/
                        "%.1f",
                        base_dsc->id2 * 1.0f);
```

Reply:
In the suggestion we're missing the extra characters for the dot and the decimal number, example `label_draw_dsc->text` is 10, the final label we want is `10.0` (length is 4), if we use `lv_strlen("10")` it is equal to 2, we need to add 2 to it.

Reply:
See 48edf28, I had to add 3 to it instead of 2, I don't know why :smile_cat: , otherwise it didn't worked as expected.